Calibration/HcalCalibAlgos/test/HcalIsoTrkAnalyzer.cc

```diff
@@ -396,21 +449,75 @@ void HcalIsoTrkAnalyzer::analyze(edm::Event const& iEvent, edm::EventSetup const
   t_TracksProp = trkCaloDirections.size();
   t_ietaAll->clear(); t_ietaGood->clear();
 #ifdef EDM_ML_DEBUG
-  edm::LogInfo("HcalIsoTrack") << "# of propagated tracks " << t_TracksProp
-			       << " out of " << t_Tracks << " with Trigger "
-			       << ignoreTrigger_;
+  edm::LogVerbatim("HcalIsoTrack") << "# of propagated tracks " << t_TracksProp
+				   << " out of " << t_Tracks << " with Trigger "
+				   << ignoreTrigger_;
 #endif
+
   //Trigger
-  t_trgbits->clear();
-  t_TracksSaved = 0;
+  t_trgbits->clear(); t_hltbits->clear();
+  t_TracksSaved = t_TracksLoose = t_TracksTight = 0;
+  t_L1Bit       = true;
   t_TrigPass    = false;
-  for (unsigned int i=0; i<trigNames_.size(); ++i) t_trgbits->push_back(false);
+  for (unsigned int i=0; i<trigNames_.size(); ++i) {
+    t_trgbits->emplace_back(false);
+    t_hltbits->emplace_back(false);
+  }
+
+  //L1
+  l1GtUtils_->retrieveL1(iEvent,iSetup,tok_alg_);
+  const std::vector<std::pair<std::string, bool> > finalDecisions = l1GtUtils_->decisionsFinal();
+  for (auto decision : finalDecisions) {
+    if (decision.first.find(l1TrigName_.c_str()) != std::string::npos) {
+      t_L1Bit = decision.second;
+      break;
+    }
+  }
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HcalIsoTrack") << "Trigger Information for " << l1TrigName_
+				   << " is " << t_L1Bit << " from a list of "
+				   << finalDecisions.size() << " decisions";
+#endif
+  
+  //HLT
+  edm::Handle<edm::TriggerResults> triggerResults;
+  iEvent.getByToken(tok_trigRes_, triggerResults);
+  if (triggerResults.isValid()) {
+    const edm::TriggerNames & triggerNames = iEvent.triggerNames(*triggerResults);
+    const std::vector<std::string> & triggerNames_ = triggerNames.triggerNames();
```

Comment:
trailing _ are for class member variables

Reply:
done